### PR TITLE
Use SafeMath in worklets

### PR DESCRIPTION
**Pull request by brunobar79**
Fixes APP-1494

## What changed (plus any additional context for devs)
1) Using SafeMath ONLY inside worklet functions where tokens amount are used for
- operations (mul / div)
- comparisons (equal, gt, lt, etc)

2) Removed unused functions for formatting

## Screen recordings / screenshots
https://github.com/rainbow-me/rainbow/assets/1247834/e576664c-7ffa-46fe-bf9b-a7e86e669932

## What to test
Number related stuff in swaps.

### CODE REVIEW

File: src/__swaps__/safe-math/SafeMath.ts [hunk 1]
```diff
@@ -186,10 +188,24 @@ export function powWorklet(base: string | number, exponent: string | number): st
   if (isZeroWorklet(exponentStr)) {
     return '1';
   }
+  if (exponentStr === '1') {
+    return baseStr;
+  }
+
+  if (lessThan(exponentStr, 0)) {
```
Comment (walmat) [inline src/__swaps__/safe-math/SafeMath.ts:195]: this needs to be a worklet function

Comment (brunobar79) [reply src/__swaps__/safe-math/SafeMath.ts]: Fixed!

**Review by walmat — APPROVED** [review]
ty king 🥇

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 2]
```diff
@@ -867,9 +874,7 @@ export function useSwapInputsController({
           const inputAmount = Number(
             valueBasedDecimalFormatter({
               amount:
-                inputNativePrice > 0
-                  ? Number(inputValues.value.inputNativeValue) / inputNativePrice
-                  : Number(inputValues.value.outputAmount),
+                inputNativePrice > 0 ? divWorklet(inputValues.value.inputNativeValue, inputNativePrice) : inputValues.value.outputAmount,
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:877]: `greaterThanWorklet` for the initial `inputNativePrice > 0` comparison?

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 3]
```diff
@@ -882,12 +887,10 @@ export function useSwapInputsController({
...
               ...values,
               inputAmount,
-              inputNativeValue: Number(inputValues.value.inputNativeValue),
+              inputNativeValue: inputValues.value.inputNativeValue,
               outputAmount:
-                outputNativePrice > 0
-                  ? Number(inputValues.value.outputNativeValue) / outputNativePrice
-                  : Number(inputValues.value.inputAmount),
-              outputNativeValue: Number(inputValues.value.outputNativeValue),
+                outputNativePrice > 0 ? divWorklet(inputValues.value.outputNativeValue, outputNativePrice) : inputValues.value.inputAmount,
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:892]: same here for `outputNativePrice > 0` comparison

File: src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts [hunk 4]
```diff
@@ -29,6 +29,8 @@ import {
 } from '@/resources/assets/externalAssetsQuery';
 import { ethereumUtils } from '@/utils';
 import { queryClient } from '@/react-query';
+import { divWorklet, equalWorklet, greaterThanWorklet, mulWorklet, toFixedWorklet } from '@/__swaps__/safe-math/SafeMath';
+import { userAssetsStore } from '@/state/assets/userAssets';
```
Comment (jinchung) [inline src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts:33]: looks like this is imported but unused

**Review by jinchung — APPROVED** [review]
🌮

**Pull request merged**